src/display/display_utils.js

```diff
@@ -66,19 +67,41 @@ class DOMCMapReaderFactory {
     this.isCompressed = isCompressed;
   }
 
-  fetch({ name, }) {
+  async fetch({ name, }) {
     if (!this.baseUrl) {
-      return Promise.reject(new Error(
+      throw new Error(
         'The CMap "baseUrl" parameter must be specified, ensure that ' +
-        'the "cMapUrl" and "cMapPacked" API parameters are provided.'));
+        'the "cMapUrl" and "cMapPacked" API parameters are provided.');
     }
     if (!name) {
-      return Promise.reject(new Error('CMap name must be specified.'));
+      throw new Error('CMap name must be specified.');
+    }
+    const url = this.baseUrl + name + (this.isCompressed ? '.bcmap' : '');
+    const compressionType = (this.isCompressed ? CMapCompressionType.BINARY :
+                                                 CMapCompressionType.NONE);
+
+    if ((typeof PDFJSDev !== 'undefined' && PDFJSDev.test('MOZCENTRAL')) ||
+        isFetchSupported() && isValidFetchUrl(url, document.baseURI)) {
+      return fetch(url).then(async (response) => {
+        if (!response.ok) {
+          throw new Error(response.statusText);
+        }
+        let cMapData;
+        if (this.isCompressed) {
+          cMapData = new Uint8Array(await response.arrayBuffer());
+        } else {
+          cMapData = stringToBytes(await response.text());
+        }
+        return { cMapData, compressionType, };
+      }).catch((reason) => {
+        throw new Error(`Unable to load ${this.isCompressed ? 'binary ' : ''}` +
+                        `CMap at: ${url}`);
+      });
     }
-    return new Promise((resolve, reject) => {
-      let url = this.baseUrl + name + (this.isCompressed ? '.bcmap' : '');
 
-      let request = new XMLHttpRequest();
+    // The Fetch API is not supported.
```

Comment:
Is there a reason why we don't use a polyfill for `fetch` instead of having manual fallback logic? Since it's not too much code I don't really have a strong preference, but it looks like `babel-preset-env` can do this (https://stackoverflow.com/questions/44908044/babel-not-polyfilling-fetch-when-using-babel-preset-env) and there is also https://github.com/github/fetch.

Reply:
Given the (somewhat) advanced usage of `fetch` found in https://github.com/mozilla/pdf.js/blob/master/src/display/fetch_stream.js it's not clear to me if a polyfill would generally work well.

The main reason for *not* attempting to polyfill though, is that as you can see in the added checks `fetch` only works for the `http`/`https` protocols, whereas `XMLHttpRequest` can work for e.g. the `file`/`ftp`/... protocols as well.